### PR TITLE
WT-9221 Generate call log for open_session()

### PR DESCRIPTION
**Pull request by Sean04**
Added a function for generating call log entry for open session.

### CODE REVIEW

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

File: src/conn/conn_api.c [hunk 1]
```diff
@@ -1283,6 +1283,12 @@ __conn_open_session(WT_CONNECTION *wt_conn, WT_EVENT_HANDLER *event_handler, con
     WT_ERR(__wt_open_session(conn, event_handler, config, true, &session_ret));
     *wt_sessionp = &session_ret->iface;
 
+#ifdef HAVE_CALL_LOG
+    if (F_ISSET(conn, WT_CONN_CALL_LOG_ENABLED)) {
```
Comment (Siddhartha8899) [inline src/conn/conn_api.c:1287]: I am not sure we need to check this flag here.

If for some reason the call_log setup fails such that this flag is not set, then the connection is closed. Hence, the `open_session` function would inaccessible.

File: src/conn/conn_api.c [hunk 2]
```diff
@@ -1283,6 +1283,12 @@ __conn_open_session(WT_CONNECTION *wt_conn, WT_EVENT_HANDLER *event_handler, con
...
+        WT_TRET(__wt_call_log_open_session(session_ret, ret));
```
Comment (Siddhartha8899) [inline src/conn/conn_api.c:1288]: Should we call this after `err:`, as we did for connection?

**Review by Siddhartha8899 — CHANGES_REQUESTED** [review]

File: src/conn/conn_api.c (hunk 1 above)
Comment (Sean04) [reply src/conn/conn_api.c:1287]: Good point, I previously had that there when I was calling __wt_call_log_open_session for the internal sessions.

**Review by Siddhartha8899 — APPROVED** [review]

**Pull request merged**